src/util/ThreadedMailbox.cc

```diff
@@ -116,29 +130,75 @@ namespace litecore { namespace actor {
     }
 
     void ThreadedMailbox::enqueue(std::function<void()> f) {
+        beginLatency();
         retain(_actor);
-        if (push(f))
+        const auto wrappedBlock = [f, SELF]
+        {
+            endLatency();
+            beforeEvent(false);
+            safelyCall(f);
+            afterEvent();
+        };
+
+        if (push(wrappedBlock))
             reschedule();
     }
 
     void ThreadedMailbox::enqueueAfter(delay_t delay, std::function<void()> f) {
         if (delay <= delay_t::zero())
             return enqueue(f);
 
-        auto actor = _actor;
-        retain(actor);
+        beginLatency();
         _delayedEventCount++;
+        retain(_actor);
 
-        auto timer = new Timer([f, actor, this]
+        auto timer = new Timer([f, this]
         { 
-            enqueue(f);
-            _delayedEventCount--;
-            release(actor);
+            const auto wrappedBlock = [f, SELF]
+            {
+                endLatency();
+                beforeEvent(true);
+                safelyCall(f);
+                afterEvent();
+            };
+            enqueue(wrappedBlock);
         });
+
         timer->autoDelete();
         timer->fireAfter(chrono::duration_cast<Timer::duration>(delay));
     }
 
+    void ThreadedMailbox::beforeEvent(bool fromEnqueueAfter)
+    {
+        beginBusy();
+#if ACTORS_TRACK_STATS
+        ++_callCount;
+        if(eventCount() > _maxEventCount) {
+            _maxEventCount = eventCount();
+        }
+#endif
+
+        if(fromEnqueueAfter) {
+            --_delayedEventCount;
```

Comment:
Same comment — this seems like the wrong time to decrement. Why not at the start of afterEvent()?

Reply:
My mistake — I should have said "at the _end_ of afterEvent", i.e. after `_actor->afterEvent()` is called.